tests/smoke/features/step_definitions/app.files.ts

```diff
@@ -178,3 +175,196 @@ When(
     }
   }
 )
+
+When(
+  '{string} creates new versions of the following file(s)',
+  async function (this: World, stepUser: string, stepTable: DataTable): Promise<void> {
+    const actor = this.actorContinent.get({ id: stepUser })
+    const { allFiles: allFilesPage } = new FilesPage({ actor })
+    const uploadInfo = stepTable.hashes().reduce((acc, stepRow) => {
+      const { to, resource } = stepRow
+
+      if (!acc[to]) {
+        acc[to] = []
+      }
+
+      acc[to].push(this.fileContinent.get({ name: resource }))
+
+      return acc
+    }, {})
+
+    for (const folder of Object.keys(uploadInfo)) {
+      await allFilesPage.uploadFiles({ folder, files: uploadInfo[folder], newVersion: true })
+    }
+  }
+)
+
+When(
+  '{string} declines following resource(s)',
+  async function (this: World, stepUser: string, stepTable: DataTable) {
+    const actor = this.actorContinent.get({ id: stepUser })
+    const { sharedWithMe: sharedWithMePage } = new FilesPage({ actor })
+    const shares = stepTable.raw().map((f) => f[0])
+
+    for (const share of shares) {
+      await sharedWithMePage.declineShare({ name: share })
+    }
+  }
+)
+
+Then(
+  '{string} ensures that the following resource(s) exist',
+  async function (this: World, stepUser: string, stepTable: DataTable) {
+    const actor = this.actorContinent.get({ id: stepUser })
+    const { allFiles: allFilesPage } = new FilesPage({ actor })
+    const resources = stepTable.raw().map((f) => f[0])
+
+    await allFilesPage.navigate()
+
+    for (const resource of resources) {
+      const resourceExist = await allFilesPage.resourceExist({ name: resource })
+
+      if (!resourceExist) {
+        throw new Error(`resource wasn't find: "${resource}"`)
+      }
+
+      await allFilesPage.navigate()
+    }
+  }
+)
+
+Then(
+  '{string} ensures that the following resource(s) do(es) not exist',
+  async function (this: World, stepUser: string, stepTable: DataTable) {
```

Comment:
can't we group this with '{string} ensures that the following resource(s) exist', and check via regex if it's a inverse exist or exist